inst/examples/dataone.R

```diff
@@ -0,0 +1,39 @@
+library(httr)
+library(purrr)
+library(tibble)
+library(contenturi)
+
+resp <- httr::GET("https://cn.dataone.org/cn/v2/query/solr/?q=datasource:*KNB&fl=identifier,checksum,checksumAlgorithm,replicaMN&wt=json")
```

Comment:
Cool to see dataONE an entry into the package. Did you intend to include this in the pull request?

Reply:
right, this belongs on the `dataone` branch, that was just me being sloppy on branches.  It's just an example for proof-of-principle / exploration.  Here's what I've learned so far, would love to compare notes! 
 - DataONE already will return hashes for all it's objects already, but the different repos use different algorithms.  MD5 is the most common.
- the member node download URLs aren't always stable, some more stable than others.  The most recent are usually (but not always -- the central node may have to notify the member nodes if they move content and don't update the list) returned by that solr query.   
- Some of the dataone objects are quite large.  There's about 2.5 million objects listed from the above solr query, ~ 4 are over a TB each, the list there totals about 61 TB.  This is because it doesn't include the largest objects on DataONE, which have metadata files on record but not download URLs, I believe the total size of data one is over 1 PB.  Even hashing the 61 TB independently will probably take some time.  

Anyway, would love to brainstorm more about this, including if there's a way to make use of the pre-computed hashes that aren't sha256 -- can move into a separate issue thread.

Reply:
Thanks for sharing your insights on DataONE - especially the wide range of object sizes - having this size estimate helps to have more detailed conversations about the data network, especially around the resources (e.g., money, time, bandwidth, storage) needed to find, access, move, and archive the associated content.  

The md5 hashes or other hashes, I can imagine using hash://md5/1234.... notation . Happy to contribute to brainstorm especially if there's a clear use case around some content registered and accessed through the DataONE network.